prototypes/shared/src/avisi/apps/tech_testing_ground/prototypes/shared/platforms/monday/api_wrapper.clj

```diff
@@ -0,0 +1,133 @@
+        date-obj (java.time.LocalDateTime/now)]
+    {:date (.format formatter date-obj)}))
+
+(defn add-item [board-id {:item/keys [name status]}]
+  (let [status-index (.indexOf item-statuses status)
+        column-values (json/write-str
+                        (cond-> {:date4 (current-date-monday-format)}
+                          (< -1 status-index) (assoc :status {:index status-index})))]
+    (let
+      [{{:keys [id]} :create_item}
+         (sent-query
+           {:query
+              "mutation ($board_id: Int!, $item_name: String, $column_values: JSON) { create_item(board_id: $board_id, item_name: $item_name, column_values: $column_values){ id }}"
+            :variables
+              {:board_id board-id
+               :item_name name
+               :column_values column-values}})]
+      {:item/id id
+       :item/name name
+       :item/status status})))
+
+(defn update-item
+  [board-id
+   {:item/keys [id name status]
+    :as item}]
+  (let
+    [column-values (json/write-str
+                     {:name name
+                      :status {:label status}})
+     {{:keys [id name]
+       [{status :text}] :column_values}
+        :change_multiple_column_values}
+       (sent-query
+         {:query
+            "mutation ($item_id: Int $board_id: Int!, $column_values: JSON!) { change_multiple_column_values(item_id: $item_id, board_id: $board_id, column_values: $column_values ) {id name column_values(ids: \"status\") {text}}}"
+          :variables
+            {:board_id board-id
+             :item_id id
+             :column_values column-values}})]
+    {:item/id id
+     :item/name name
+     :item/status status}))
+
+(defn delete-item
+  [_
+   {:item/keys [id]
+    :as item}]
+  (let
+    [{{:keys [id name]
+       [{status :text}] :column_values}
+        :delete_item}
+       (sent-query
+         {:query
+            "mutation($item_id: Int) { delete_item(item_id: $item_id) {id name column_values(ids: \"status\") {text}}}"
+          :variables {:item_id id}})]
+    {:item/id id
+     :item/name name
+     :item/status status}))
+
+(comment
+  (get-items 3990111892)
+  (add-item
+    3990111892
+    {:item/name "from the repl"
+     :item/status "Stuck"})
+  (update-item
+    3990111892
+    {:item/id 4255515535
+     :item/name "New repl name!!!"
+     :item/status "Working on it"})
+  (delete-item 3990111892 {:item/id 4254394618}))
+
+(comment
+  ; queries for use with eql->gql lib, but no addded value was found in doing this so the decision has been made
+  ; to stick with pure gql-queries. These queries are documented here for now in case reasons arise to move to the lib
+  ; after all.
+  [{:boards [:id :name]}]
+  [{(:boards {:ids [~board-id]}) [{:items [:id :name]}]}]
+  [{('create_item
+     {:board_id ~board-id
+      :item_name ~item-name
+      :column_values ~column-values})
+      [:id]}]
+  [{('delete_item {:item_id ~item-id}) [:id]}])
```

Comment:
Ik zou deze keuze heroverwegen omdat je nu strings aan elkaar aan het plakken bent kan je issue krijgen met dingen zoals `"{..}"` als input

Reply:
I dont need to concatenate strings as I use graph-qls native way of substituting variables. I think it's fine for this usecase and going with the clojure-datastructure option only starts to make sense when you need to dynamically compose the query itself. I might need a rational for this decision later though.